src/Dependencies/Collections/SegmentedArray`1+PrivateMarshal.cs

```diff
@@ -14,5 +16,14 @@ internal static class PrivateMarshal
         /// <inheritdoc cref="SegmentedCollectionsMarshal.AsSegments{T}(SegmentedArray{T})"/>
         public static T[][] AsSegments(SegmentedArray<T> array)
             => array._items;
+
+        public static SegmentedArray<T> AsSegmentedArray(T[][] segments)
+        {
+            var length = segments.Length == 0 ? 0 : segments[segments.Length - 1].Length;
```

Comment:
💭 Given the advanced use case, would it be better to make length an argument?

Reply:
I could go either way. I assume you would prefer it so I'll go ahead and switch to that.

Reply:
➡️ Now simplified